### ISSUE TITLE
NullReferenceException: Premature deployment of feature

### ISSUE THREAD

**Issue by Vectorial1024** [Vectorial1024/Desynchronized#23]
This bug is introduced in the v1.6.0 update. It only occurs occasionally during normal social interactions. The cause is currently unknown.

Normally, there is a probability for social interactions to transmit news. When transmitting news, a tale-news is selected from the initiator's list of known news, and sent to the receiver's list of known news.

Example error log follows.

```
System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.PawnRelationUtility+<GetRelations>c__Iterator0.MoveNext () [0x00000] in <filename unknown>:0 
  at RimWorld.PawnRelationUtility.GetMostImportantRelation (Verse.Pawn me, Verse.Pawn other) [0x00000] in <filename unknown>:0 
  at Desynchronized.TNDBS.Utilities.NewsImportanceExtensionHelper.GetSocialProximityScoreForOther (Verse.Pawn subject, Verse.Pawn other) [0x00000] in <filename unknown>:0 
  at Desynchronized.TNDBS.TaleNewsPawnDied.CalculateNewsImportanceForPawn (Verse.Pawn pawn, Desynchronized.TNDBS.TaleNewsReference reference) [0x00000] in <filename unknown>:0 
  at Desynchronized.TNDBS.TaleNewsReference.RecalculateNewsImportance () [0x00000] in <filename unknown>:0 
  at Desynchronized.TNDBS.TaleNewsDatabase.RecalculateTaleNewsImportance () [0x00000] in <filename unknown>:0 
  at Desynchronized.TNDBS.TaleNewsDatabase.Tick () [0x00000] in <filename unknown>:0 
  at RimWorld.Planet.WorldObjectsHolder.WorldObjectsHolderTick () [0x00000] in <filename unknown>:0 
  at RimWorld.Planet.World.WorldTick () [0x00000] in <filename unknown>:0 
  at Verse.TickManager.DoSingleTick () [0x00000] in <filename unknown>:0 
Verse.Log:Error(String, Boolean)
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()
```

**Comment by Vectorial1024** [comment]
Very quickly, the cause has been found.

I probably introduced "forgetfulness" in v1.6.0 unintentionally. Supposedly, if a news is completely forgotten, the details will be purged. This results in the relevant pawns to become null. However, when I attempt to update the news importance score, I didn't check the null presence, and thus caused errors.

The fix is in the upcoming v1.6.1, but we need more testing to be sure.

**Comment by Vectorial1024** [comment]
Cannot reproduce and no more reports, so closed.
